library/asn1write.c

```diff
@@ -99,12 +99,15 @@ int mbedtls_asn1_write_len( unsigned char **p, unsigned char *start, size_t len
     return( MBEDTLS_ERR_ASN1_INVALID_LENGTH );
 }
 
-int mbedtls_asn1_write_tag( unsigned char **p, unsigned char *start, unsigned char tag )
+int mbedtls_asn1_write_tag( unsigned char **p, unsigned char *start, int tag )
```

Comment:
Again, this is a breaking change to the interface, so we can't accept the PR as it is. You'll need to create a new function.